frame/contracts/src/lib.rs

```diff
@@ -829,6 +827,8 @@ pub mod pallet {
 		StorageDepositNotEnoughFunds,
 		/// More storage was created than allowed by the storage deposit limit.
 		StorageDepositLimitExhausted,
+		/// The storage deposit limit needs to be specified.
+		StorageDepositLimitRequired,
```

Comment:
why do we need this?

Reply:
This is because of the reason on the previous point. Do you think we should use another existing error instead?

Reply:
as mentioned below, is this still relevant when #13565 lands and we get a fallback limit?

Reply:
In #13565 `limit` is still an `Option` within the [`new` function](https://github.com/paritytech/substrate/pull/13565/files#diff-c1c6c9bef0c70e17c8a7f3124d2e6f87646ae1a1f0ac314c80e9ee3b1d51bf57L333) therefore we  have to handle it.
If at some point the `limit` type changes from being an `Option<BalanceOf<T>>` to be a `BalanceOf<T>` we can very well remove it